### PR TITLE
Add planning defense phase 

### LINKED ISSUE #15: Add planning defense phase
With #14 implemented, we can have a planning state and game state. In the planning state we can place towers (or spawn units), and when the round starts, the state changes and the round is played. 

For now, let's just ignore the units we spawn in the planning phase (i.e. just delete them), and play the game according to roundSimulation.json.

### PR DESCRIPTION
**Pull request by AndreasWintherMoen**
Closes #15 

This adds the basic infrastructure for a tower placement feature. So far we just spawn a tower at a position, but this is erased when we actually start the game, and new actual towers are loaded based on the round simulation json file. 

This one was a bit problematic related to the publish/subscribe patterns we've used. For example if a Renderable is spawned from another Renderable's render function, the list of Renderables in Renderer is modified while we loop through it. The same happens with Updater and ClickPublisher (which is a new class added in this PR). The solution I came up with was to have a list of new subscribers to be added and removed, and then add or remove them at the end of the frame after performing the subscriber functions. There may be a better solution to the problem, but this solution seems to work, although it is a bit ugly. 

I also added some new manager classes and reorganized some folders, so even though the planning defense phase isn't fully implemented, I figured we should do a PR and synchronize branches. There will be some merge conflicts!

### CODE REVIEW

File: client/core/src/com/cows/game/managers/FunctionDelayer.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+object FunctionDelayer {
+    private val thisFrameFunctions = mutableListOf<() -> Unit>()
+    private val nextFrameFunctions = mutableListOf<() -> Unit>()
+
+    fun invokeFunctionAtEndOfThisFrame(function: () -> Unit) = thisFrameFunctions.add(function)
+
+    fun invokeFunctionAtEndOfNextFrame(function: () -> Unit) = nextFrameFunctions.add(function)
+
+    fun invokeRegisteredFunctions() {
+        thisFrameFunctions.forEach { println(it); it.invoke() }
```
Comment (blauks) [inline client/core/src/com/cows/game/managers/FunctionDelayer.kt:12]: Do we really need this?

File: client/core/src/com/cows/game/hud/HUDManager.kt [hunk 2]
```diff
@@ -1,23 +1,71 @@
...
+    }
+
+    private fun createPlanningAttackButtons() {
+        val startGameButton = Button("HUD/start-button.png", { onStartGame.invoke() })
+
+        buttons.add(startGameButton)
+    }
+
+    private fun createActiveGameButtons() {
+
```
Comment (blauks) [inline client/core/src/com/cows/game/hud/HUDManager.kt:69]: Add TODO?

**Review by blauks — CHANGES_REQUESTED** [review]
"Kan man trykke på veien?" - @Nicolaad 21.03.22 12:10. Could be looked at.

Other than that, looks good!

**Comment by AndreasWintherMoen** [comment]
> "Kan man trykke på veien?" - @Nicolaad 21.03.22 12:10. Could be looked at.
> 
> Other than that, looks good!

Fixed!

**Pull request merged**